### PR TITLE
Issue289

### PR DESCRIPTION
**Pull request by darrenleeweber**
First attempt to address one error in the display of a Revs object, to address #289.  Unable to get the argo-stage object data into a fixture to completely test this fix.  The additional spec does test just this helper method.  Although travis CI, hound, coveralls etc. can evaluate this PR to some extent, the PR branch might have to be pushed to argo-stage to better evaluate how it behaves with the data noted in #289.

### CODE REVIEW

File: spec/helpers/value_helper_spec.rb [hunk 1]
```diff
@@ -31,4 +31,25 @@
...
+      field = 'originInfo_date_created_tesim'
+      value = ['1966', 'anything_after_first_element_ignored']
+      document = SolrDocument.new({ field => value })
+      args = { document: document, field: field }
+      expect(helper.value_for_originInfo_date_created_tesim(args))
+        .to eq(value.first)
+    end
+    it 'does normalize a time stamp string' do
+      now_utc = Time.now.utc.to_s
+      now_loc = Time.parse(now_utc).localtime.strftime '%Y.%m.%d %H:%M%p'
```
Comment (darrenleeweber) [inline spec/helpers/value_helper_spec.rb]: How are you supposed to know the offset when this code is run and could be run on any machine anywhere?  The whole point of not specifying the offset is to use the local machine time.  Sorry, don't agree with houndci on this one.

File: app/helpers/value_helper.rb [hunk 2]
```diff
@@ -69,9 +69,14 @@ def links_to_collections(args)
     end.join('<br>').html_safe
   end
 
+  # Discussed removing this method because the field should not be modified.
   def value_for_originInfo_date_created_tesim(args)
-    val = Time.parse(args[:document][args[:field]].first)
+    field_value = args[:document][args[:field]].first
+    # Try to parse the string as a time value
+    val = Time.parse(field_value)
```
Comment (darrenleeweber) [inline app/helpers/value_helper.rb]: The very next line of this code does use the `localtime` zone of the machine running the code.  Good enough?  (Guess houndci doesn't look ahead?)

File: spec/helpers/value_helper_spec.rb (hunk 1 above)
Comment (cbeer) [inline spec/helpers/value_helper_spec.rb]: I believe `Time#in_time_zone` is the environment-aware local time zone.

File: app/helpers/value_helper.rb (hunk 2 above)
Comment (cbeer) [inline app/helpers/value_helper.rb]: Is it? I think the behavior rubocop is trying to discourage here is using `Time.parse` with a poorly-specified timezone reference, to catch differences like:

```ruby
# Using the system time zone (PT)
irb(main):003:0> Time.parse('2015-01-01')
=> 2015-01-01 00:00:00 -0800

# Using the application time zone (UTC)
irb(main):004:0> Time.zone.parse('2015-01-01')
=> Thu, 01 Jan 2015 00:00:00 UTC +00:00
```

See also this pretty nice guide to time handling in rails:  http://danilenko.org/2012/7/6/rails_timezones/

File: lib/tasks/argo.rake [hunk 3]
```diff
@@ -261,9 +261,12 @@ namespace :argo do
   end
 
   def load_order_files(fedora_files)
+    data_path = File.expand_path('../../../fedora_conf/data/', __FILE__)
```
Comment (mejackreed) [inline lib/tasks/argo.rake]: Seems like this might be part of another PR or a different commit. One looking at this commit down the road might wonder why this task was changed as part of this bug fix.

File: app/helpers/value_helper.rb [hunk 4]
```diff
@@ -69,9 +69,14 @@ def links_to_collections(args)
...
     val.localtime.strftime '%Y.%m.%d %H:%M%p'
+  rescue
```
Comment (mejackreed) [inline app/helpers/value_helper.rb]: If we are going to use the `rescue` approach, it might be good to rescue from the specific error that is raising rather than `StandardError`.

File: app/helpers/value_helper.rb [hunk 5]
```diff
@@ -69,9 +69,14 @@ def links_to_collections(args)
     end.join('<br>').html_safe
   end
 
+  # Discussed removing this method because the field should not be modified.
   def value_for_originInfo_date_created_tesim(args)
```
Comment (mejackreed) [inline app/helpers/value_helper.rb]: Since this method isn't doing anything specifically related to `orginInfo_date_created_tesim` it would be nice to see it renamed and added explicitly using `helper_method` in `catalog_controller.rb` https://github.com/sul-dlss/argo/blob/develop/app/controllers/catalog_controller.rb#L49 https://github.com/projectblacklight/blacklight/wiki/Configuration---Results-View#helper-method

File: spec/helpers/value_helper_spec.rb (hunk 1 above)
Comment (darrenleeweber) [inline spec/helpers/value_helper_spec.rb]: Can't find that one, i.e.
```
$ ruby -v
ruby 2.2.3p173 (2015-08-18 revision 51636) [x86_64-linux]
$ pry
[1] pry(main)> Time.in_time_zone
NoMethodError: undefined method `in_time_zone' for Time:Class
[2] pry(main)> Time.now.in_time_zone
NoMethodError: undefined method `in_time_zone' for 2015-12-10 08:40:18 -0800:Time
```

Comment (mejackreed) [inline spec/helpers/value_helper_spec.rb]: `#in_time_zone` is an ActiveSupport thing, I believe. Try within the context of the rails console.

```sh
$ rails c
[1] pry(main)> Time.now.in_time_zone
=> Thu, 10 Dec 2015 16:46:40 UTC +00:00
```

File: lib/tasks/argo.rake (hunk 3 above)
Comment (mejackreed) [inline lib/tasks/argo.rake]: See now this is in another commit... :ok_hand:

Comment (mejackreed) [inline lib/tasks/argo.rake]: Also was this causing an error?

Comment (darrenleeweber) [inline lib/tasks/argo.rake]: Yes, understood.  I didn't have the energy to be perfect after a less than perfect day at the keyboard ;-)  This file was touched because I wanted a blank line in the `load_order` file to separate out the fixtures included for working on #289, which have been subsequently removed for several reasons, primarily because they are not representative of the argo-stage data and secondarily because it may not be good to add a new fixture data file for every bug and lastly because the spec code may suffice to test the method modifications in this PR.  But thanks for encouraging perfection and I'm open to suggestions on how to work around this if it's really important.

File: app/helpers/value_helper.rb (hunk 4 above)
Comment (darrenleeweber) [inline app/helpers/value_helper.rb]: Good point.  Without knowing what the data is, it's not obvious what the exceptions may be.  Will explore this in revised specs and add specific exception classes.

File: lib/tasks/argo.rake (hunk 3 above)
Comment (mejackreed) [inline lib/tasks/argo.rake]: Nope just curious about the intent here as I didn't understand. :ok:

File: app/helpers/value_helper.rb (hunk 2 above)
Comment (darrenleeweber) [inline app/helpers/value_helper.rb]: It looks like argo does not explicitly set the app time zone, so `Time.zone` is the default UTC.  Can we assume that this data should be parsed as a UTC value?

Comment (cbeer) [inline app/helpers/value_helper.rb]: I think it's safe to assume the application should be set to the correct time zone (and whether it is or not  is a separate question that, indeed, should be answered..)

File: app/helpers/value_helper.rb (hunk 5 above)
Comment (darrenleeweber) [inline app/helpers/value_helper.rb]: Thanks for the tip and refs.  This PR should use this!  Will revise it.

**Comment by darrenleeweber** [comment]
After further discussion, this solution will be replaced with an alternative that removes the helper method entirely.
